### PR TITLE
Add logger option to GCPPubSubStrategy

### PR DESCRIPTION
**Pull request by epicatization**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nestjs-gcp-pubsub",
-  "version": "0.2.1",
+  "version": "0.2.2",
```
Comment (iamolegga) [inline package.json:3]: no need to add this to a commit. Releases are made separately

Comment (epicatization) [reply package.json]: okay, fixed

File: src/server.ts [hunk 2]
```diff
@@ -16,12 +16,13 @@ export class GCPPubSubStrategy
...
   implements CustomTransportStrategy
 {
-  protected readonly logger = new Logger(GCPPubSubStrategy.name);
+  protected readonly logger: Logger;
   private readonly subscriptions: Record<string, Subscription> = {};
 
   constructor(private readonly opts: GCPPubSubServerOptions) {
     super();
     this.deserializer = opts.deserializer ?? new JSONDeserializer();
+    this.logger = opts.logger ?? new Logger(GCPPubSubStrategy.name);
```
Comment (iamolegga) [inline src/server.ts:25]: I'm wondering if there are any official recommendations on properly doing this. Could you please point me to how logger overrides are made in official transports, please?

Comment (epicatization) [reply src/server.ts:25]: sure, i ll look around if there are any recommendations

Comment (iamolegga) [reply src/server.ts:25]: Thanks, I think this should be a pretty common part of all transports. If you won't find any then let's merge it, I just want to be sure

Comment (iamolegga) [reply src/server.ts:25]: No problem. Sure, logger implementation substitution should work just fine here

**Pull request closed without merging**